addons/web/views/report_templates.xml

```diff
@@ -406,12 +406,12 @@
         </div>
 
         <div t-attf-class="footer o_company_#{company.id}_layout {{report_type != 'pdf' and 'mt-auto'}}">
-            <div class="o_footer_content row border-top pt-2">
-                <div class="col-8">
+            <div t-attf-class="o_footer_content border-top pt-2 {{'d-flex' if is_html_empty(company.report_header) else 'row' }}">
```

Comment:
Thanks for your changes! The visual result looks good, I'm just curious why do we need either a `d-flex` or a `row` because the `row` class also applies a d-flex by default. Was it creating some unwanted margins / padding on the html report rendering?

Reply:
I tried with only the row and adding the `me-2` after the `div.flex-grow-1` but rendering in pdf fail with the page number 
![image](https://github.com/user-attachments/assets/0d08b853-b855-4cf5-8fdc-3b7e7f97424a)

I don't have subtility between d-flex and row but it seems slightly different. Feel free to edit the PR directly if you want to test or correct something

Reply:
Indeed it doesn't work 🤔 PDF rendering is capricious. I added a commit on top of your to apply the spacing between the footer & page number and a `text-nowrap`. Since the page number is rendered only in PDF we have to add an extra condition for the `me-2` else it would offset the content in HTML.

I'm waiting to test on runbot but locally on my machine it was rendering well, feel free to squash the commit if you agree with the changes / if it looks good on your setup as well

Reply:
@amoyaux  I just tested on runbot and the `text-nowrap` is offsetting the page numbers vertically. Let's just go for your initial solution and drop the spacing I'm suggesting. I'm dropping my commit. Sorry for the struggle.
![image](https://github.com/user-attachments/assets/8298f41f-dc5f-423b-925f-4d222cef08c3)